### PR TITLE
Reogranize constants

### LINKED ISSUE #939: Reorganize constants
Currently there are lot of loose constants around. Some of them are required to pass down as a parameter not to create code couplings. 
I think we should reorganize constants to a single location similarly as done in `ConfigurationKeys.cs`.

Some examples to fix: `MeterSettings.cs` `TracerSettings.cs` `ConfigurationSourceExtensions.cs`
Some examples of common constants: `,` `;` `:` `none` 

_Issue mentioned by @pellared https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/937#discussion_r921980215_

### PR DESCRIPTION
**Pull request by RassK**
## Why

Fixes #939

## What

Moves all necessary constants to `Constants.cs` with similar tree system as `ConfigurationKeys.cs`.

### CODE REVIEW

File: src/OpenTelemetry.AutoInstrumentation.StartupHook/OpenTelemetry.AutoInstrumentation.StartupHook.csproj [hunk 1]
```diff
@@ -4,4 +4,10 @@
     <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
   </PropertyGroup>
 
+  <ItemGroup>
+    <Compile Include="..\OpenTelemetry.AutoInstrumentation\Constants.cs" Link="Constants.cs">
+      <CopyToOutputDirectory>Never</CopyToOutputDirectory>
```
Comment (pjanotti) [inline src/OpenTelemetry.AutoInstrumentation.StartupHook/OpenTelemetry.AutoInstrumentation.StartupHook.csproj:9]: Is this really needed for a `Compile` item?

File: src/OpenTelemetry.AutoInstrumentation/Constants.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+            /// <summary>
+            /// Default delimiter for textual representation of multi-valued settings.
+            /// </summary>
+            public const char Separator = ',';
+
+            /// <summary>a
+            /// Delimiter for textual representation of settings that contains multiple
+            /// fully qualified .NET names, e.g.: assembly or type names. Comma is part
+            /// of a fully qualified name and can't be used here.
```
Comment (pjanotti) [inline src/OpenTelemetry.AutoInstrumentation/Constants.cs:42]:
```suggestion
            /// Delimiter for textual representation of settings that may contain multiple
            /// fully qualified .NET names, e.g.: assembly or type names, that already
            /// use commas as separators.
```

**Review by pjanotti — COMMENTED** [review]
One Q and one suggestion for a comment.

File: src/OpenTelemetry.AutoInstrumentation.StartupHook/OpenTelemetry.AutoInstrumentation.StartupHook.csproj (hunk 1 above)
Comment (RassK) [reply src/OpenTelemetry.AutoInstrumentation.StartupHook/OpenTelemetry.AutoInstrumentation.StartupHook.csproj]: Fixed.

**Review by nrcventura — APPROVED** [review]

**Review by pjanotti — APPROVED** [review]

**Pull request merged**